# 0x05. Python - Exceptions

## Table of contents
Files | Description
----- | -----------
[0-safe_print_list.py](./0-safe_print_list.py) | Python function that prints x elements of a list and passes when there is an exception
[1-safe_print_integer.py](./1-safe_print_integer.py) | Python function prints an integer with "{:d}".format() and handles exception
[2-safe_print_list_integers.py](./2-safe_print_list_integers.py) | Python function that prints the first x elements of a list and only integers and handles exception
[3-safe_print_division.py](./3-safe_print_division.py) | Python function that divides 2 integers and prints the result and handles exception
[4-list_division.py](./4-list_division.py) | Python function that divides element by element 2 lists and handles exception
[5-raise_exception.py](./5-raise_exception.py) | Python function that raises a type exception.
[6-raise_exception_msg.py](./6-raise_exception_msg.py) | Python function that returns a set of common elements in two sets
[100-safe_print_integer_err.py](./100-safe_print_integer_err.py) | Python function function that prints an integer. The value can be any type (integer, string, etc.) The integer should be printed followed by a new line. Returns True if value has been correctly printed (it means the value is an integer) otherwise, returns False and prints in stderr the error precede by Exception:
[101-safe_function.py](./101-safe_function.py) | Python function that returns a set of common elements in two sets
[102-magic_calculation.py](./102-magic_calculation.py) |Python function def magic_calculation(a, b): that does exactly the same as some Python bytecode.
[103-python.c](./103-python.c) | C functions that print some basic info about Python lists, Python bytes an Python float objects.